### PR TITLE
feature (refs ADO-17195): create ParagraphFactoryInterface.php

### PR DESCRIPTION
**Pull request by jalalifardemos**
Ticket: https://www.dev.diplanung.de/DefaultCollection/BOP-HH/_workitems/edit/17195

Description: create a factory for paragraph entity

core: https://github.com/demos-europe/demosplan-core/pull/3290
cockpit: https://github.com/demos-europe/demosplan-addon-xbauleitplanung/pull/34

### CODE REVIEW

File: src/Contracts/Handler/ParagraphHandlerInterface.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+interface ParagraphHandlerInterface
+{
+    public function administrationDocumentNewHandler(string $procedure, string $category, array $data, $elementId);
+
+    public function administrationDocumentEditHandler($procedureId, $data, $elementId);
+
+    public function administrationDocumentDeleteHandler(ProcedureInterface $procedure, ElementsInterface $element);
+
+}
```
Comment (breierDemos) [inline src/Contracts/Handler/ParagraphHandlerInterface.php:16]: Empty line before closing bracket can be deleted.

**Review by breierDemos — CHANGES_REQUESTED** [review]
Code Review only

It seams that both files have missing a new line after the closing bracket.
Changelog entry is missing.

File: src/Contracts/Handler/ParagraphHandlerInterface.php (hunk 1 above)
Comment (jalalifardemos) [reply src/Contracts/Handler/ParagraphHandlerInterface.php:15]: done in [d121525](https://github.com/demos-europe/demosplan-addon/pull/116/commits/d1215254b109407013cdbd0065755b6e05ed0146)

**Review by MoritzMandler — APPROVED** [review]
lgtm

**Review by bakridemos — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## UNRELEASED
 
+## v0.39 (2024-08-06)
```
Comment (mussbach) [inline CHANGELOG.md:5]: Please do not change version and content in one PR as you usually cannot know for sure that this PR is the next version.

File: composer.json [hunk 3]
```diff
@@ -61,5 +61,5 @@
   "scripts": {
     "phpstan": "phpstan analyse"
   },
-  "version": "v0.38"
+  "version": "v0.39"
```
Comment (mussbach) [inline composer.json:64]: Please do not change version and content in one PR as you usually cannot know for sure that this PR is the next version

**Pull request merged**